### PR TITLE
R4R: add auto migrate task for ibc_tx_latest and cronjon for ibc_tx

### PR DESCRIPTION
**Pull request by weichang-bianjie**

### CODE REVIEW

File: src/monitor/ibc_monitor.service.ts [hunk 1]
```diff
@@ -42,7 +43,7 @@ export class IbcMonitorService {
         this.ibcTxModel = await this.connection.model(
```
Comment (jinbiao-bianjie) [inline src/monitor/ibc_monitor.service.ts:43]: ibcTxModel -> ibcTxLatestTableModel

File: src/service/ibc_tx.service.ts [hunk 2]
```diff
@@ -120,19 +138,180 @@ export class IbcTxService {
...
+            const txModel = await this.connection.model(
+                'txModel',
+                TxSchema,
+                `sync_${dcChainID}_tx`,
+            );
+            let dcTxData = await txModel.queryTxByHash(dcTxHash)
+
+            if (dcTxData?.length && dcTxData[dcTxData?.length - 1]?.events_new) {
+                const {connect, ackData} = await this.getConnectByRecvPacketEventsNews(dcTxData[dcTxData?.length - 1]);
+                dcConnect = connect
```
Comment (jinbiao-bianjie) [inline src/service/ibc_tx.service.ts:249]: gets event_new for  corresponding msg index

File: src/task/ibc_tx_latest_migrate_task.service.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            status: IbcTxStatus.SETTING,
+            limit: MaxMigrateBatchLimit})
+        const Txs = await this.ibcTxLatestModel.queryTxsLimit(limit,1)
+        const batchTxs = [...Txs,...settingTxs]
+
+        const session = await this.connection.startSession()
+        session.startTransaction()
+        try {
+            for (const one of batchTxs) {
+                await this.ibcTxModel.insertManyIbcTx([one],session)
```
Comment (jinbiao-bianjie) [inline src/task/ibc_tx_latest_migrate_task.service.ts:69]: insertMany?

File: src/task/ibc_tx_latest_migrate_task.service.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        const settingTxs = await this.ibcTxLatestModel.queryTxsByStatusLimit({
+            status: IbcTxStatus.SETTING,
+            limit: MaxMigrateBatchLimit})
+        const Txs = await this.ibcTxLatestModel.queryTxsLimit(limit,1)
+        const batchTxs = [...Txs,...settingTxs]
+
+        const session = await this.connection.startSession()
+        session.startTransaction()
+        try {
+            for (const one of batchTxs) {
```
Comment (jinbiao-bianjie) [inline src/task/ibc_tx_latest_migrate_task.service.ts:68]: one -> ibcTx

File: src/service/ibc_tx.service.ts [hunk 5]
```diff
@@ -120,19 +137,195 @@ export class IbcTxService {
...
+                }
+            }
+        }
+        return {
+            ack,
+            dcConnect
+        }
+    }
+
+    async getIbcTxDetail(query) {
```
Comment (kaifei-bianjie) [inline src/service/ibc_tx.service.ts:272]: param `query` rename to `txHash`

File: src/service/ibc_tx.service.ts [hunk 6]
```diff
@@ -120,19 +137,195 @@ export class IbcTxService {
...
+        const ibcTxLatest = await this.ibcTxLatestModel.queryTxDetailByHash(query)
```
Comment (kaifei-bianjie) [inline src/service/ibc_tx.service.ts:273]: pseudocode as follow
```
if ibcTxLatest is not empty
  set value of ibcTxDetail
else
 query txs from ibc_tx
 set value of ibcTxDetail
```

File: src/task/ibc_tx_latest_migrate_task.service.ts (hunk 4 above)
Comment (kaifei-bianjie) [inline src/task/ibc_tx_latest_migrate_task.service.ts:68]: use batch insert, don't insert one by one

File: src/task/ibc_tx_latest_migrate_task.service.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+                await this.ibcTxModel.insertManyIbcTx([one],session)
+                await this.ibcTxLatestModel.deleteIbcTx(one,session)
```
Comment (kaifei-bianjie) [inline src/task/ibc_tx_latest_migrate_task.service.ts:70]: use delete many, don't delete one by one

File: src/task/ibc_tx_latest_migrate_task.service.ts [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        await this.startMigrate(migrateCnt)
+        Logger.log("ibc migrate have finished,migrate count:",migrateCnt)
+    }
+
+    async startMigrate(migrateCount):  Promise<void> {
+        //todo migrate start condition value with timestamp random
+        const value =  (Math.floor(new Date().getTime() / 1000)%10)*10000
+        if (migrateCount > value) {
+            // migrate max bitch count limit
+            if (migrateCount > MaxMigrateBatchLimit) {
```
Comment (kaifei-bianjie) [inline src/task/ibc_tx_latest_migrate_task.service.ts:51]: max handle record num is MaxMigrateBatchLimit?

**Review by kaifei-bianjie — CHANGES_REQUESTED** [review]

**Review by jinbiao-bianjie — APPROVED** [review]

**Pull request merged**